ext/drreg/drreg.c

```diff
@@ -1719,37 +2714,82 @@ drreg_event_restore_state(void *drcontext, bool restore_memory,
         return true; /* fault not in cache */
     for (reg = DR_REG_START_GPR; reg <= DR_REG_STOP_GPR; reg++)
         spilled_to[GPR_IDX(reg)] = MAX_SPILLS;
+#ifdef SIMD_SUPPORTED
+    for (reg = DR_REG_APPLICABLE_START_SIMD; reg <= DR_REG_APPLICABLE_STOP_SIMD; reg++)
+        spilled_simd_to[SIMD_IDX(reg)] = MAX_SIMD_SPILLS;
+    for (slot = 0; slot < ops.num_spill_simd_slots; ++slot)
+        simd_slot_use[slot] = DR_REG_NULL;
+#endif
+
     LOG(drcontext, DR_LOG_ALL, 3,
         "%s: processing fault @" PFX ": decoding from " PFX "\n", __FUNCTION__,
         info->raw_mcontext->pc, pc);
     instr_init(drcontext, &inst);
+    instr_init(drcontext, &next_inst);
     while (pc < info->raw_mcontext->pc) {
         instr_reset(drcontext, &inst);
+        instr_reset(drcontext, &next_inst);
         prev_pc = pc;
         pc = decode(drcontext, pc, &inst);
-
-        if (is_our_spill_or_restore(drcontext, &inst, &spill, &reg, &slot, &offs)) {
+        if (pc == NULL) {
+            LOG(drcontext, DR_LOG_ALL, 3, "%s @" PFX " \n", __FUNCTION__, prev_pc,
+                "PC decoding returned NULL during state restoration");
+            instr_free(drcontext, &inst);
+            instr_free(drcontext, &next_inst);
+            return true;
+        }
+        decode(drcontext, pc, &next_inst);
```

Comment:
Also check for pc == NULL!

Reply:
This is already being checked no?

Reply:
What about checking the success of the second decode?

Reply:
Done, let me know if you want it in a separate function